### PR TITLE
[Core] Fix get_actor consistency on ray.kill

### LINKED ISSUE #20129: [Bug] get_actor right after ray.kill returns the dead actor.
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/ray/issues) and found no similar issues.

### Ray Component

Ray Core

### What happened + What you expected to happen

Currently, calling `get_actor` right after `ray.kill` can return the stale dead actor. Since ray.kill is blocking, and it is passing through GCS, it should theoretically never happen, so there must be a bug. 

### Versions / Dependencies

master

### Reproduction script

https://github.com/ray-project/ray/blob/f65cc72b4ccb43e13c04f1273248f4f86ed85cae/python/ray/tests/test_actor_advanced.py#L1327

Basically, we should fix this. 

### Anything else

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

This fixes the issue where `get_actor` returns the dead actor handle after it is used after `ray.kill`.

The problem was that we cache the actor handle that's obtained by `get_actor` in memory, and that's invalidated when GCS "publishes the dead actor info". There's usually natural gap between ray.kill is called & dead actor info is published.

We solve the issue by invalidating the cache after ray.kill is returned (so we have no gap here).

In the CI environment, it is not happening commonly because ray.kill is blocking, and before it is responded, the actor data is published. But it could be more common once we turn on grpc based pubsub. 

## Related issue number

Closes https://github.com/ray-project/ray/issues/20129

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: src/ray/core_worker/core_worker.cc [hunk 1]
```diff
@@ -2074,7 +2075,9 @@ Status CoreWorker::KillActor(const ActorID &actor_id, bool force_kill, bool no_r
       cb(Status::Invalid(stream.str()));
     }
   });
-  return f.get();
+  const auto &status = f.get();
+  actor_manager_->InvalidateActorHandleCache(actor_id);
```
Comment (scv119) [inline src/ray/core_worker/core_worker.cc:2079]: this leaks the implementation detail. should we rename it to a more general function, such as (OnActorKilled) or somethign?

File: src/ray/protobuf/core_worker.proto [hunk 2]
```diff
@@ -55,6 +55,12 @@ message ActorHandle {
 
   // How many times tasks may be retried on this actor if the actor fails.
   int64 max_task_retries = 9;
+
+  // The name of the actor.
+  string name = 10;
+
+  // The namespace that this actor belongs to.
+  string ray_namespace = 11;
```
Comment (scv119) [inline src/ray/protobuf/core_worker.proto:63]: looks we are inconsistent about namespace vs ray_namespace; any reason why?

File: src/ray/core_worker/core_worker.cc (hunk 1 above)
Comment (rkooo567) [reply src/ray/core_worker/core_worker.cc:2079]: Hmm yeah. I thought about InvalidateActorHandle, and noticed it didn't make sense, so I just changed this. OnActorKilled sounds good.

File: src/ray/protobuf/core_worker.proto (hunk 2 above)
Comment (rkooo567) [reply src/ray/protobuf/core_worker.proto:63]: The existing RPC messages seem to use ray_namespace, so I followed it. I think namespace was a reserved keyword or something (I tried namespace at first, and didn't work for some reasons).

**Comment by rkooo567** [comment]
cc @scv119

**Review by scv119 — APPROVED** [review]

**Comment by rkooo567** [comment]
Failed test unrelated/

**Pull request merged**